src/CMakeLists.txt

```diff
@@ -9,6 +9,7 @@ add_subdirectory(solvers)
 add_subdirectory(integrators)
 add_subdirectory(coefficients)
 add_subdirectory(drivers)
+add_subdirectory(infrastructure)
```

Comment:
I actually like this less-used name than `utils` because it is more descriptive.